packages/react-sdk/src/components/ImportWhiteboardDialog/pdfImportUtils.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2024 Nordeck IT + Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  getDocument,
+  GlobalWorkerOptions,
+  PDFDocumentProxy,
+  PDFPageProxy,
+} from 'pdfjs-dist';
+import { RenderParameters } from 'pdfjs-dist/types/src/display/api';
+
+// This is a weird oddity of pdfjs. It requires the worker to be loaded from a URL.
+export async function initPDFJs() {
+  if (!Object.prototype.hasOwnProperty.call(process.env, 'SSR')) {
+    console.log('Loading pdf.worker.mjs in webpack');
+    // @ts-expect-error -- There are no types for this file
+    await import('pdfjs-dist/webpack');
+  } else {
+    console.log('Loading pdf.worker.mjs in vitejs');
+    // @ts-expect-error -- There are no types for this file
+    const pdfJSWorker = await import('pdfjs-dist/build/pdf.worker.mjs?url');
+    GlobalWorkerOptions.workerSrc = pdfJSWorker.default;
+  }
+}
+
+export interface PDFImportResult {
+  data: ArrayBuffer;
+  mimeType: string;
+  size: number;
+  width: number;
+  height: number;
+}
+
+export async function loadPDF(file: ArrayBuffer): Promise<PDFDocumentProxy> {
+  const pdfLoadingTask = await getDocument(file);
+  const pdf = await pdfLoadingTask.promise;
+  return pdf;
+}
+
+export async function renderPDFToImages(
+  pdf: PDFDocumentProxy,
+  desiredWidth: number = 1920,
+  desiredHeight: number = 1080,
+): Promise<PDFImportResult[]> {
+  const images = [];
+  for (let i = 1; i <= pdf.numPages; i++) {
+    const page = await pdf.getPage(i);
+    const imageData = await renderPDFPageToImage(
+      page,
+      desiredWidth,
+      desiredHeight,
+    );
+    images.push(imageData);
+  }
+  return images;
+}
+
+async function renderPDFPageToImage(
+  page: PDFPageProxy,
+  desiredWidth: number = 1920,
+  desiredHeight: number = 1080,
+): Promise<PDFImportResult> {
+  const viewport = page.getViewport({ scale: 1 });
+  let scale = desiredWidth / viewport.width;
+
+  // Cap the scaledViewport to be a maximum of desiredWidth*desiredHeight
+  let scaledViewport = page.getViewport({ scale: scale });
+  if (scaledViewport.height > desiredHeight) {
+    scale = desiredHeight / viewport.height;
+    scaledViewport = page.getViewport({ scale: scale });
+  } else if (scaledViewport.width > desiredWidth) {
+    scale = desiredWidth / viewport.width;
+    scaledViewport = page.getViewport({ scale: scale });
+  }
+
+  // Support HiDPI-screens.
+  const outputScale = window.devicePixelRatio || 1;
```

Comment:
@HarHarLinks Could you receck this with the current version of this branch if it still is an issue?

Reply:
it works, but when set to `1` it means I always only get "1080p quality" imports on my 4K screen. Since at this time we can assume users will have 4K screens frequently enough to matter, I think we should work out the numbers (is `outputScale=2` enough? not sure) so the result will be "4K quality". We might also use a `max(window.devicePixelRatio, X)` so if someone should happen to use it on a 5K Mac with 30% browser zoom, they get the "6.5K quality" import they need to be flawless.

That is, unless we can import pdf as svg instead?

Reply:
I dont think we can get an svg from PDF.

Will revisit the scaling.